js/xt.js

```diff
@@ -0,0 +1,1072 @@
+            const networks = {};
+            let depositEnabled = undefined;
+            let withdrawEnabled = undefined;
+            for (let j = 0; j < rawNetworks.length; j++) {
+                const rawNetwork = rawNetworks[j];
+                const networkId = this.safeString (rawNetwork, 'name');
+                const network = this.safeNetwork (networkId);
+                const depositRaw = this.safeString (rawNetwork, 'deposit');
+                let deposit = false;
+                if (depositRaw === 1) {
+                    deposit = true;
+                    depositEnabled = true;
+                }
+                const withdrawRaw = this.safeString (rawNetwork, 'withdraw');
+                let withdraw = false;
+                if (withdrawRaw === 1) {
+                    withdraw = true;
+                    withdrawEnabled = true;
+                }
+                networks[network] = {
+                    'info': rawNetwork,
+                    'id': networkId,
+                    'network': network,
+                    'fee': undefined,
+                    'deposit': deposit,
+                    'withdraw': withdraw,
+                    'limits': {
+                        'withdraw': {
+                            'min': undefined,
+                            'max': undefined,
+                        },
+                        'deposit': {
+                            'min': undefined,
+                            'max': undefined,
+                        },
+                    },
+                };
+            }
+            result[code] = {
+                'info': entry,
+                'code': code,
+                'id': currencyId,
+                'precision': undefined,
+                'name': name,
+                'active': true,
+                'deposit': depositEnabled,
+                'withdraw': withdrawEnabled,
+                'networks': networks,
+                'fee': undefined,
+                'limits': {
+                    'amount': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                },
+            };
+        }
+        return result;
+    }
+
+    safeNetwork (networkId) {
+        if (networkId === undefined) {
+            return undefined;
+        } else {
+            return networkId.toLowerCase ();
+        }
+    }
+
+    async fetchMarkets (params = {}) {
+        /**
+         * @method
+         * @name xt#fetchMarkets
+         * @description retrieves data on all markets for xt
+         * @param {object} params extra parameters specific to the exchange api endpoint
+         * @returns {[object]} an array of objects representing market data
+         */
+        // let type = undefined;
+        // [ type, params ] = this.handleMarketTypeAndParams ('fetchMarkets', undefined, params);
+        const query = this.omit (params, 'type');
```

Comment:
this does not seem complete, where's the spot loading?

Reply:
Oops commented out the spot part. Since the spot & USDT-M exchange ids are the same I only load the spot markets by default

Reply:
Oops the spot part is commented out